src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmRpoServiceImpl.java

```diff
@@ -53,4 +81,54 @@ public void updateArmRpoStatus(ArmRpoExecutionDetailEntity armRpoExecutionDetail
     public ArmRpoExecutionDetailEntity saveArmRpoExecutionDetailEntity(ArmRpoExecutionDetailEntity armRpoExecutionDetailEntity) {
         return armRpoExecutionDetailRepository.save(armRpoExecutionDetailEntity);
     }
+
+    @Override
+    public void reconcileArmRpoCsvData(ArmRpoExecutionDetailEntity armRpoExecutionDetailEntity, List<File> csvFiles,
+                                       List<MasterIndexFieldByRecordClassSchema> headerColumns) {
+        List<Integer> csvEodList = new ArrayList<>();
+        Duration duration = Duration.between(armRpoExecutionDetailEntity.getCreatedDateTime(), currentTimeHelper.currentOffsetDateTime());
+        ObjectRecordStatusEntity armRpoPending = EodHelper.armRpoPendingStatus();
+        StringBuilder errorMessage = new StringBuilder("Failure during ARM RPO CSV Reconciliation: ");
+
+        ArmAutomatedTaskEntity armAutomatedTaskEntity = armAutomatedTaskRepository.findByAutomatedTask_taskName(ADD_ASYNC_SEARCH_RELATED_TASK_NAME)
+            .orElseThrow(() -> new ArmRpoException(errorMessage.append("Automated task ProcessE2EArmRpoPending not found.").toString()));
+
+        OffsetDateTime now = currentTimeHelper.currentOffsetDateTime();
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities = externalObjectDirectoryRepository.findByStatusAndIngestionDate(
+            armRpoPending,
+            now.minusHours(armAutomatedTaskEntity.getRpoCsvEndHour() + (long) duration.toHoursPart()),
+            now.minusHours(armAutomatedTaskEntity.getRpoCsvStartHour() + (long) duration.toHoursPart()));
+        String[] headers = headerColumns.stream()
```

Comment:
Ordering doesnt matter as we extract the header using Column name.

Reply:
So, given this csv data (from `src/test/resources/Tests/arm/rpo/armRpoCsvData.csv):
```
Record Class,Archived Date,Client Identifier,Contributor,Record Date,ObjectId,ParentId,Channel,MaxChannels,ObjectType,CaseNumbers,FileType,Checksum,TranscriptRequest,TranscriptType,TranscriptUrgency,Comments,UploadedBy,HearingDate,CreatedDateTime,StartDateTime,EndDateTime,Courthouse,Courtroom
DARTS,,1,Cth1-Ctr1,,1,1,1,4,Media,T123456,MP2,1234,,,,,testUser,,,,,Cth1,Ctr1
```

Then try changing the order of the strings in the `CSV_HEADER` array in your test - if I move `"Client Identifier"` from 3rd position to 4th position, then the code picks up the `Cth1-Ctr1` value, so it seems the ordering is significant. The actual header values in the csv file don't seem to matter - by supplying the headers I think we're explicitly telling the parser which column name belongs to which data column. So there's potential to pick up the wrong column data if the header ordering is different from what logically exists in the csv file.

Do we even need to manually set the headers? If the csv file already contains the header values (as the `armRpoCsvData.csv` does), can we just let the parser automatically read them with `builder.setHeader()`? See the JavaDoc on `CSVFormat.setHeader()`:

```
        /**
         * Sets the header to the given values. The header can be parsed automatically from the input file with:
         *
         * <pre>
         * builder.setHeader();
         * </pre>
         *
         * or specified manually with:
         *
         * <pre>{@code
         * builder.setHeader("name", "email", "phone");
         * }</pre>
         * <p>
         * The header is also used by the {@link CSVPrinter}.
         * </p>
         *
         * @param header the header, {@code null} if disabled, empty if parsed automatically, user-specified otherwise.
         * @return This instance.
         */
        public Builder setHeader(final String... header) {
            this.headers = CSVFormat.clone(header);
            return this;
        }
```

If we can do that it would allow us to eliminate the `List<MasterIndexFieldByRecordClassSchema> headerColumns` param from the `reconcileArmRpoCsvData()` method signature, simplify the code and avoid the ordering pitfalls.